src/main/java/Duke/Tasks/Task.java

```diff
@@ -0,0 +1,100 @@
+package duke.tasks;
+
+import duke.errors.InvalidCommandException;
+
+/**
+ * The type Task.
+ */
+public abstract class Task {
+    /**
+     * The Description.
+     */
+    protected String description;
+    /**
+     * The Done.
+     */
+    protected String done;
+
+    /**
+     * Instantiates a new Task.
+     *
+     * @param description the description
+     * @param done        the done
+     */
+    public Task(String description, String done) {
+        this.description = description;
+        this.done = done;
+    }
+
+    /**
+     * Getter method to get Task Description
+     *
+     * @return Task Description
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Takes in an array of Task values to form into a String to be stored on database
+     *
+     * @param arr String Array of Task values
+     * @return String to be saved to database
+     * @throws Exception the exception
+     */
+    public static String stringFormat(String[] arr) throws Exception {
+        String s;
+        switch (arr[0]) {
+        case "T":
+            s = String.format("%s,%s,%s\n", arr[0], arr[1], arr[2]);
+            break;
+        case "E":
+            s = String.format("%s,%s,%s,%s\n", arr[0], arr[1], arr[2], arr[3]);
+            break;
+        case "D":
+            s = String.format("%s,%s,%s,%s\n", arr[0], arr[1], arr[2], arr[3]);
+            break;
+        default:
+            throw new InvalidCommandException();
+        }
+        return s;
+    }
+
+    /**
+     * Sets Status of Task to be done or not.
+     *
+     * @param checked String "0" to represent false, and "1" for true
```

Comment:
Hello 😄 .  Could you use a boolean for the property done instead? If there are more than 2 states, you could consider using a enum 🔢 .  Otherwise all good 👍